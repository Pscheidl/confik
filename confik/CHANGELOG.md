# Changelog

## Unreleased

- Attributes that receive expressions (`default` and `from`) now need to be unquoted. E.g.:

  ```diff
  - struct MyStruct(#[config(default = "\"Hello World\"") String)
  + struct MyStruct(#[config(default = "Hello World") String)

  - struct MyStruct(#[config(default = "5usize") usize)
  + struct MyStruct(#[config(default = 5usize) usize)
  ```

## 0.7.0

- Initial release.
