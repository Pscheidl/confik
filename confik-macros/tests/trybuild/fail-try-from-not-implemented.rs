//! Check the error returned when `TryFrom`/`TryInto` is not implemented for the given types.

use confik::Configuration;

#[derive(Debug, Configuration)]
struct Config {
    #[confik(try_from = String)]
    param: Foo,
}

#[derive(Debug, Default, serde::Deserialize, confik::Configuration)]
struct Foo(String);

fn main() {}
