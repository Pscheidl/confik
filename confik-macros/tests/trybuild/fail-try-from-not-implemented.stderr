error[E0277]: the trait bound `Foo: From<std::string::String>` is not satisfied
 --> tests/trybuild/fail-try-from-not-implemented.rs:5:17
  |
5 | #[derive(Debug, Configuration)]
  |                 ^^^^^^^^^^^^^ the trait `From<std::string::String>` is not implemented for `Foo`
  |
  = note: required for `std::string::String` to implement `Into<Foo>`
  = note: required for `Foo` to implement `TryFrom<std::string::String>`
  = note: required for `std::string::String` to implement `std::convert::TryInto<Foo>`
  = note: this error originates in the derive macro `Configuration` (in Nightly builds, run with -Z macro-backtrace for more info)
